make_travis_yml_2.hs

```diff
@@ -71,29 +73,43 @@ versionNumbers (Version vn _) = vn
 
 #endif
 
-putStrLnErr :: MonadIO m => String -> m a
-putStrLnErr  m = liftIO $ hPutStrLn stderr ("*ERROR* " ++ m) >> exitFailure
+data MakeTravisOutput
+    = Success [String] [String]
+    | Failure [String]
+    deriving (Eq, Show)
+
+instance Monoid MakeTravisOutput where
+    mempty = Success [] []
+    Failure err1 `mappend` Failure err2 = Failure $ err1 `mappend` err2
+    Failure err1 `mappend` Success err2 _ = Failure $ err1 `mappend` err2
+    Success err1 _ `mappend` Failure err2 = Failure $ err1 `mappend` err2
+    Success l1 o1 `mappend` Success l2 o2 =
+        Success (mappend l1 l2) (mappend o1 o2)
 
-putStrLnWarn, putStrLnInfo :: MonadIO m => String -> m ()
-putStrLnWarn m = liftIO $ hPutStrLn stderr ("*WARNING* " ++ m)
-putStrLnInfo m = liftIO $ hPutStrLn stderr ("*INFO* " ++ m)
+type YamlWriter m a = MaybeT (WriterT MakeTravisOutput m) a
 
--- putStrLns :: [String] -> IO ()
--- putStrLns = putStr . unlines
+putStrLnErr :: Monad m => String -> YamlWriter m a
+putStrLnErr m = do
+    lift . tell $ Failure ["*ERROR* " ++ m]
+    mzero
```

Comment:
is this the reason for `MaybeT`?

Reply:
Yeah, it was the simplest way I could think of to preserve the termination on errors.

Reply:
I'll add a note to that effect to YamlWriter.